.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -2655,9 +2616,99 @@ stages:
             baseImage: debian
             command: "CheckBuildLogsForErrors"
 
+- stage: nuget_installer_smoke_tests_arm64
+  condition: and(succeeded(), eq(variables['isScheduledBuild'], 'False'))
+  dependsOn: [dotnet_tool, package_windows, generate_variables, master_commit_id]
+  variables:
+    masterCommitId: $[ stageDependencies.master_commit_id.fetch.outputs['set_sha.master']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [linux]
+
+  - job: linux
+    timeoutInMinutes: 60 #default value
+    strategy:
+      matrix: $[ stageDependencies.generate_variables.generate_variables_job.outputs['generate_variables_step.nuget_installer_smoke_tests_arm64_matrix'] ]
+    variables:
+      smokeTestAppDir: "$(System.DefaultWorkingDirectory)/tracer/test/test-applications/regression/AspNetCoreSmokeTest"
+    pool:
+      name: aws-arm64-auto-scaling
+
+    steps:
+    - template: steps/clone-repo.yml
+      parameters:
+        masterCommitId: $(masterCommitId)
+
+    - task: DownloadPipelineArtifact@2
+      displayName: Download distribution nuget
+      inputs:
+        artifact: distribution-nuget-package
+        path: $(smokeTestAppDir)/artifacts
+
+    - task: DownloadPipelineArtifact@2
+      displayName: Download Datadog.Trace package
+      inputs:
+        artifact: nuget-packages
+        path: $(smokeTestAppDir)/artifacts
+
+    - script: |
+        mkdir -p tracer/build_data/snapshots
+        mkdir -p tracer/build_data/logs
+      displayName: create test data directories
+
+    - script: |
+        docker-compose -p ddtrace_$(Build.BuildNumber) build \
+          --build-arg DOTNETSDK_VERSION=$(dotnetCoreSdkLatestVersion) \
+          --build-arg RUNTIME_IMAGE=$(runtimeImage) \
+          --build-arg PUBLISH_FRAMEWORK=$(publishFramework) \
+          --build-arg RELATIVE_PROFILER_PATH="$(relativeProfilerPath)" \
+          nuget-smoke-tests
+      env:
+        dockerTag: $(dockerTag)
+      displayName: docker-compose build nuget-smoke-tests
+      retryCountOnTaskFailure: 3
```

Comment:
nit - could be moved to a dedicated step as well as you do it twice. But we could do it if we include windows at some point obv.

Reply:
This is different depending on which test we're running, e.g. installer tests vs nuget tests, and Windows will be different, so I don't think moving it really helps